packaging/macos/tests/test_discover_agent_version.sh

```
@@ -190,6 +190,191 @@ t_unknown_connector() {
+}
+
+t_probe_json_version_records_malformed_when_log_set() {
+  local dir; dir="$(mktest_tmp)"
+  local cfg="${dir}/pkg.json"
+  local log="${dir}/errors.log"
+  : > "${log}"
+  # Malformed: unclosed object.
+  printf '{"version":"1.0.0"' > "${cfg}"
+  local out
+  DC_DISCOVERY_ERRORS_LOG="${log}" \
+    DC_INSTALLER_TARGET_USER="alice" \
```

Shellcheck (0.11.0)

[warning] DC_INSTALLER_TARGET_USER appears unused. Verify use (or export if used externally).

(SC2034)